pkg/kubeapiserver/options/api_enablement.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"strings"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/apiserver/pkg/server"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
+	"k8s.io/apiserver/pkg/server/resourceconfig"
+	serverstore "k8s.io/apiserver/pkg/server/storage"
+)
+
+// APIEnablementOptions holds the APIEnablement options.
+// It is a wrap of generic APIEnablementOptions.
+type APIEnablementOptions struct {
+	GenericAPIEnablement *genericoptions.APIEnablementOptions
```

Comment:
why a pointer?

Reply:
Just for NewAPIEnablementOptions returns a pointer. Otherwise, it seems both ok.

Reply:
in the options a pointer suggests it can be set to nil to disable a component. Not the case here.